src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+#include "overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope.h"
+
+s32 gDbgCamEnabled = false;
+u8 D_801D0D54 = false;
+
+// bss
+s16 sTransitionFillTimer;
+Input D_801F6C18;
+FbDemoStruct sTrnsnUnk;
+u16* D_801F6D0C;
+s32 gTrnsnUnkState;
+VisMono sVisMono;
+Color_RGBA8_u32 gVisMonoColor;
+Struct_80140E80 D_801F6D38;
+Struct_80140E80* D_801F6D4C;
+HiresoStruct sHireso;
+u8 sBombersNotebookOpen;
+u8 sMotionBlurStatus;
+
+typedef enum {
+    /* 0 */ MOTION_BLUR_STOPPED,
+    /* 1 */ MOTION_BLUR_INITALIZING,
+    /* 2 */ MOTION_BLUR_RUNNING
+} MotionBlurStatus;
+
+void Play_DrawMotionBlur(PlayState* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+    s32 alpha;
+    Gfx* gfx;
+    Gfx* dlistHead;
+
+    if (R_MOTION_BLUR_PRIORITY_ENABLED) {
+        alpha = R_MOTION_BLUR_PRIORITY_ALPHA;
+
+        if (sMotionBlurStatus == MOTION_BLUR_STOPPED) {
+            sMotionBlurStatus = MOTION_BLUR_INITALIZING;
+        }
+    } else if (R_MOTION_BLUR_ENABLED) {
+        alpha = R_MOTION_BLUR_ALPHA;
+
+        if (sMotionBlurStatus == MOTION_BLUR_STOPPED) {
+            sMotionBlurStatus = MOTION_BLUR_INITALIZING;
+        }
+    } else {
+        alpha = 0;
+        sMotionBlurStatus = MOTION_BLUR_STOPPED;
+    }
+
+    if (sMotionBlurStatus != MOTION_BLUR_STOPPED) {
+        OPEN_DISPS(gfxCtx);
+
+        dlistHead = POLY_OPA_DISP;
+        gfx = Graph_GfxPlusOne(dlistHead);
+
+        gSPDisplayList(OVERLAY_DISP++, gfx);
+
+        this->pauseBgPreRender.fbuf = gfxCtx->curFrameBuffer;
+        this->pauseBgPreRender.fbufSave = this->unk_18E64;
+
+        if (sMotionBlurStatus == MOTION_BLUR_RUNNING) {
+            func_80170AE0(&this->pauseBgPreRender, &gfx, alpha);
+        } else {
+            sMotionBlurStatus = MOTION_BLUR_RUNNING;
+        }
+
+        func_801705B4(&this->pauseBgPreRender, &gfx);
+
+        gSPEndDisplayList(gfx++);
+
+        Graph_BranchDlist(dlistHead, gfx);
+
+        POLY_OPA_DISP = gfx;
+
+        CLOSE_DISPS(gfxCtx);
+    }
+}
+
+void Play_InitMotionBlur(void) {
+    R_MOTION_BLUR_ENABLED = false;
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+    sMotionBlurStatus = MOTION_BLUR_STOPPED;
+}
+
+void Play_DestroyMotionBlur(void) {
+    R_MOTION_BLUR_ENABLED = false;
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+    sMotionBlurStatus = MOTION_BLUR_STOPPED;
+}
+
+void Play_SetMotionBlurAlpha(u32 alpha) {
+    R_MOTION_BLUR_ALPHA = alpha;
+}
+
+void Play_EnableMotionBlur(u32 alpha) {
+    R_MOTION_BLUR_ALPHA = alpha;
+    R_MOTION_BLUR_ENABLED = true;
+}
+
+void Play_DisableMotionBlur(void) {
+    R_MOTION_BLUR_ENABLED = false;
+}
+
+void Play_ConvertBufferToI(void* destI, u16* srcBuf, s32 bufWidth, s32 pixelLeft, s32 pixelTop, s32 pixelRight,
+                           s32 pixelBottom, s32 type) {
+    s32 i;
+    s32 j;
+    u32 pixel;
+    u32 r;
+    u32 g;
+    u32 b;
+
+    switch (type) {
+        case 4: {
+            u8* destI4 = (u8*)destI;
```

Comment:
I wonder if this could either use an enum, or `type` could be more descriptive of being the "size" of the `I` (`I4`/`I5`/`I8`/`I16`), I feel there should be a term for that. Might be a @Thar0 question